dll/win32/shell32/utils.cpp

```diff
@@ -9,6 +9,39 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
+static HWND
+SHELL_FindStubWindow(UINT Type, LPCWSTR Path)
+{
+    for (HWND hWnd, hWndAfter = NULL;;)
+    {
+        hWnd = hWndAfter = FindWindowExW(NULL, hWndAfter, CSTUBWINDOW32_CLASSNAME, Path);
+        if (!hWnd || !Path)
+            return NULL;
+        if (GetPropW(hWnd, CStubWindow32::GetTypePropName()) == ULongToHandle(Type))
+            return hWnd;
+    }
+}
+
+HRESULT
+CStubWindow32::CreateStub(CStubWindow32 &stub, UINT Type, LPCWSTR Path, const POINT *pPt)
+{
+    if (HWND hWnd = SHELL_FindStubWindow(Type, Path))
+    {
+        ::SwitchToThisWindow(::GetLastActivePopup(hWnd), TRUE);
+        return HRESULT_FROM_WIN32(ERROR_ALREADY_EXISTS);
+    }
+    RECT rcPosition = { pPt ? pPt->x : CW_USEDEFAULT, pPt ? pPt->y : CW_USEDEFAULT, 0, 0 };
+    DWORD Style = WS_DISABLED | WS_CLIPSIBLINGS | WS_CAPTION;
+    DWORD ExStyle = WS_EX_WINDOWEDGE | WS_EX_APPWINDOW;
+    if (!stub.Create(NULL, rcPosition, Path, Style, ExStyle))
+    {
+        ERR("StubWindow32 creation failed\n");
+        return E_FAIL;
+    }
+    ::SetPropW(stub, GetTypePropName(), ULongToHandle(Type));
```

Comment:
Because it does not need to access `this`. Nobody needs to override this name. If someone else needs a stub window, they would add a new `TYPE_` to the enum in the CStubWindow32 class (it current has two users, the drive format dialog and shell propsheet dialogs).